debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/TypeRegistry.java

```diff
@@ -98,7 +98,7 @@
     private final Map<String, PostgresType> nameToType = new HashMap<>();
     private final Map<Integer, PostgresType> oidToType = new HashMap<>();
 
-    private final PostgresConnection connection;
+    private PostgresConnection connection;
```

Comment:
I'm feeling a bit uneasy about this one; why is this needed?

Reply:
Agreed, I wasn't really happy with how I hacked this part together.  It's been a while and I'm trying to recall the specifics, but I somewhat referenced the issue with this comment: https://github.com/debezium/debezium/pull/1973/files#r529883357

This change was "forced" by trying to follow the existing SqlServer pattern where the `SqlServerValueConverter` is injected into the `SqlServerConnection`'s constructor.  However, the postgres code has an additional bit of complexity via the `TypeRegistry`.

The circular dependency between the `PostgresConnection` and `TypeRegistry` caused friction along with `PostgresValueConverter`, in which I worked around by first creating the `TypeRegistry` independently of the connection so that the registry can be injected into the converter, and then injected the registry into the connection (in which the connection then calls `prime()` to do what the registry's constructor did previously).

Basically the mess I created here: https://github.com/debezium/debezium/pull/1973/files#diff-0c1ba77f25c31166ad8d358987cf467ba70689a3cd15b58f0d486959fdc99dddR77-R94 and https://github.com/debezium/debezium/pull/1973/files#diff-f2d4d5bd7e1a3d93fe13e699b7ec7ba0fb138468c6c35b1592e08449142e87a7R82-R84

Without sufficient understanding of the rest of the code I tried to keep my changes minimal as to not break anything else by attempting a larger refactoring.  Sadly, I'm lacking the time now to drive this further even if given some additional pointers on a better design.